src/api/client.js

```diff
@@ -18,10 +25,31 @@ function assetTypesFromState(assetTypes) {
   return Object.keys(assetTypes).filter(key => assetTypes[key]);
 }
 
-export function requestAssets(courseId, { page = 0, pageSize = 50, sort = 'sort', assetTypes = {} }) {
-  const assetTypesToFilter = assetTypesFromState(assetTypes);
+function getDatabaseAttributesFromAssetAttributes(sort) {
+  return assetAttributesToDatabaseAttributes[sort];
+}
+
+function objectIsNonEmpty(object) {
+  return Object.keys(object).length > 0;
+}
+
+export function requestAssets(courseId, params) {
+  const assetTypesToFilter = assetTypesFromState(params.assetTypes);
+  const sortType = getDatabaseAttributesFromAssetAttributes(params.sort);
+
+  const parameters = {
+    ...params,
+    sort: sortType,
+    asset_type: (objectIsNonEmpty(assetTypesToFilter) ? assetTypesToFilter : undefined),
+    assetTypes: undefined,
+    page_size: params.pageSize,
+    pageSize: undefined,
+  };
+
+  const requestString = Object.keys(parameters).reduce((memo, key) => { if (parameters[key]) { memo.push(`${key}=${parameters[key]}`); } return memo; }, []).join('&');
```

Comment:
Because we no longer specify defaults, we have to dynamically generate the request string depending upon the contents of the parameters object.

Reply:
For my own curiosity - is it a convention to call it `memo` when performing this sort of task?

Reply:
I don't think so? I think it's just the name Ari gave it, since she helped me with this part. I think I'll rename it to request, just so it's a little more clear.